src/data/list/repeat.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2023 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import data.list.rotate
+import data.list.big_operators.lemmas
+
+/-!
+# List of a Constant Element
+
+This file proves basic results about `list.repeat x n`, the list of `x` repeated `n` times.
+-/
+
+namespace list
+
+variables {α β : Type*} (x : α) (n : ℕ)
+
+@[simp] lemma map_repeat (f : α → β) : (repeat x n).map f = repeat (f x) n :=
+begin
+  induction n with n hn,
+  { exact rfl },
+  { exact (map_cons f x _).trans (congr_arg ((::) (f x)) hn) }
+end
+
+/-- `filter` removes either all or none of `repeat x n`. -/
+@[simp] lemma filter_repeat (p : α → Prop) [decidable_pred p] :
+  (repeat x n).filter p = if p x then repeat x n else [] :=
+begin
+  induction n with n hn,
+  { exact (if_t_t _ []).symm },
+  { split_ifs with hp; simp [hp, hn] }
+end
+
+/-- Only `x` is a member of `list.repeat x n` (unless `n = 0` which has no members). -/
+@[simp] lemma mem_repeat_iff (y : α) : y ∈ repeat x n ↔ 0 < n ∧ y = x :=
+begin
+  induction n with n hn,
+  { rw [lt_self_iff_false, false_and, repeat, mem_nil_iff] },
+  { simp [hn] }
+end
+
+lemma not_mem_repeat_zero (y : α) : y ∉ (repeat x 0) :=
+by simp_rw [mem_repeat_iff, lt_self_iff_false, false_and, not_false_iff]
```

Comment:
This proof feels like overkill, don't we have a lemma that says `repeat x 0 = []`?